packages/manager/src/features/Linodes/MigrateLinode/ConfigureForm.tsx

```diff
@@ -98,6 +104,17 @@ export const ConfigureForm = React.memo((props: Props) => {
     [backupEnabled, currentLinodeType]
   );
 
+  const linodeIsInEdgeRegion =
+    flags.gecko && currentActualRegion?.site_type === 'edge';
+
+  const filterRegions = () => {
+    if (linodeIsInEdgeRegion) {
+      // edge regions can only be migrated to other edge regions
+      return filterOutCurrentRegionAndCoreRegions(currentRegion, regions ?? []);
+    }
+    return regions?.filter((eachRegion) => eachRegion.id !== currentRegion);
+  };
```

Comment:
Maybe the flag check best belongs here?

```suggestion
const getRegions = () => {
    if (flags.gecko && linodeIsInEdgeRegion) {
      return getRegionsWithoutCurrentRegionAndCoreSites(currentRegion, regions ?? []);
    }
    return regions?.filter((eachRegion) => eachRegion.id !== currentRegion);
  };

  const filteredRegions = getRegions();
```